### PR TITLE
Remove dependency on Node.js, Bootstrap, and jQuery

### PR DESCRIPTION
**Pull request by jwnrt**
This PR allows the site to build with just Hugo, and removes all of the Node setup. Best reviewed by commit.

The current build uses Node.js for pulling dependencies and pre-processing before Hugo runs:

* Bootstrap's SASS is processed with our theming.
* JS dependencies are downloaded from NPM and bundled together.
* Everything is bundled into two `.js` and `.css` files and minified.

Hugo has support for processing SASS as well as minifying JS using [pipes](https://gohugo.io/hugo-pipes/). I've moved the SASS processing to this, but decided to just drop our JS dependencies (Bootstrap's collapse.js plus its dependency on jQuery).

We're only using this for the toggle button in the navbar. I've replaced it with this function which has the same functionality but loses the smooth sliding transition when the navbar opens and closes:

```javascript
<button /* ... */ onclick="document.getElementById('navbarCollapse').classList.toggle('show')>...</button>
```

The smooth transition could be added back in a future PR if wanted, possibly through a CSS transition or by updating to Bootstrap 5 which doesn't need the jQuery dependency and is lighter. Not including in this PR since it's already a huge change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,101 +1,44 @@
...
-### Download all frontend dependencies
-
-Before you can get started, you need to download all frontend dependencies
-(CSS and JS libraries), and Hugo. These dependencies will be installed into the
-`node_modules` directory.
-
-```bash
-yarn install
+```sh
+hugo -s site -d dist --minify --baseURL 'https://www.lowrisc.org'
```
Comment (luismarques) [inline README.md:10]: We could extract this into a `build.sh` or a Makefile target or something.

File: README.md [hunk 2]
```diff
@@ -1,101 +1,44 @@
...
-[draft or future content](https://gohugo.io/getting-started/usage/#draft-future-and-expired-content).
-To see such content in the preview, run
+The following will build the site and serve a preview locally. When running,
+Hugo will trigger rebuilds and automatically refresh your webpage when files
+change.
 
-```bash
-yarn preview
+```sh
+hugo server -s site
```
Comment (luismarques) [inline README.md:30]: Makefile `serve` target?

File: README.md [hunk 3]
```diff
@@ -1,101 +1,44 @@
...
 ```
 
-## Deploy
+Hugo supports marking content as [draft or future content][hugo-draft-future].
+To preview this content, run Hugo with the `-D` and `-F` flags:
 
-A static version of the web site can be built with:
+hugo server -s site -D -F
```
Comment (luismarques) [inline README.md:36]: Makefile `preview` target or something like that?

**Comment by luismarques** [comment]
Looking good so far. For the `collapse.js` removal, would it be possible to add a CSS transition to replace the current one?

File: README.md (hunk 1 above)
Comment (jwnrt) [reply README.md]: I've moved the Hugo directories into the top level, so the commands are now:

```sh
# Preview
hugo serve
# Preview with draft and future content
hugo serve -D -F
# Build for deployment
hugo --minify --baseURL 'https://www.lowrisc.org'
```

I agree the third target is worth a Makefile so I've added one, but at some stage if the deployment is done via GitHub CI I think the first two commands are simple enough for devs to run themselves.

**Comment by HU90m** [comment]
This made my day. It looks great.

Do we want to update the CI setup before merging?

Also, we could use `transition-property`,  `transition-duration` and `transition-timing-function` to add the pretty animation back to `navbar-toggler`. I can add this later though.

**Comment by jwnrt** [comment]
> Looking good so far. For the `collapse.js` removal, would it be possible to add a CSS transition to replace the current one?

Almost certainly, but I haven't done this before. @HU90m do you know how simple these are to add?

Alternatively, the Bootstrap 5 version of collapse.js doesn't need jQuery so would be lighter than out current Bootstrap 4 setup + jQuery, but still probably more than we need. Would also have to check the JS works with our Bootstrap 4 SASS, but maybe we could update this too.

**Comment by jwnrt** [comment]
> Do we want to update the CI setup before merging?

I will try and get this working. Currently it's starting the server fine but then doing nothing for 30 minutes and killing everything. I need to work out what it's expecting to happen and fix that.

**Comment by jwnrt** [comment]
CI fixed. Netlify wanted to serve the site itself.

**Comment by HU90m** [comment]
> Almost certainly, but I haven't done this before. @HU90m do you know how simple these are to add?

Super simple, but if we are already using bootstrap maybe avoiding custom CSS would be better. I like your idea of moving to Bootstrap 5.

This PR is already pretty big though. It might be a good idea to merge this in. Then looking at adding the animation back, possibly while updating to Bootstrap 5.

File: README.md (hunk 1 above)
Comment (jwnrt) [reply README.md]: Just to clarify, the Makefile currently covers all three commands (`make`, `make preview`, `make preview-drafts`), but if `make` is eventually handled by CI automatically then I think `make preview` and `make preview-drafts` are simple enough not to need the Makefile any more.

**Comment by jwnrt** [comment]
Website was moved to wordpress so this is moot.

**Pull request closed without merging**